packages/api-client/src/rest/rest.ts

```diff
@@ -124,6 +140,25 @@ export class RestClientImpl implements RestClient {
     return result
   }
 
+  private async checkRateLimits (response: Response): Promise<void> {
+    if (response.status === 429) {
+      // Extract rate limit information from headers
+      const retryAfter = response.headers.get('Retry-After')
+      const rateLimitReset = response.headers.get('X-RateLimit-Reset')
+      // const rateLimitLimit: string | null = response.headers.get('X-RateLimit-Limit')
+      const waitTime =
+        retryAfter != null
+          ? parseInt(retryAfter, 10)
+          : rateLimitReset != null
+            ? new Date(parseInt(rateLimitReset, 10)).getTime() - Date.now()
```

Comment:
it should not be 10 here

Reply:
Maybe do something like this; `Math.max(new Date(parseInt(rateLimitReset, Date.now())).getTime() - Date.now()), 10)`

Reply:
Do not understand you and provided may be code is some really weird stuff. Especially parseInt by Date.now() radix
`parseInt(rateLimitReset, Date.now())`

Reply:
Yeah, Somehow I decided that it is default value :)